src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogSCMListener.java

```diff
@@ -157,6 +162,28 @@ public void onCheckout(Run<?, ?> build, SCM scm, FilePath workspace, TaskListene
         }
     }
 
+    private boolean isSharedLibraryCheckout(Run<?, ?> build, FilePath workspace) {
+        if (workspace == null){
+            return false;
+        }
+        FilePath parent = workspace.getParent();
+        if (parent == null) {
+            return false;
+        }
+        String name = parent.getName();
+        if (!name.endsWith("@libs")) {
+            return false;
+        }
```

Comment:
Can this happen in the `workspace` instance (not in the parent) ?

Reply:
The parent is the common folder where all shared libraries for a pipeline are stored, while the workspace is a folder which contains a specific shared library (for example `/var/jenkins_home/workspace/test-pipeline-shared-library@libs/fbf88f8b22e51df9124d2af6a9cc72fd2567aecb8b4f0d8b40967d2a6ebbe198`).
The names look like some kind of hashes, I don't think we can get any useful clues from checking them